### PR TITLE
DOC: adapts documentation for first-time users

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
This is my first time running Dask-CHTC. I've adapted the documentation for items I'd like to see in the documentation. So far, these are:

* Adds note about installing Jupyter before running
* Only show `dask-chtc jupyter run lab` instead of that and the unuseful `jupyter lab`.
* Renames "Example Notebook" to "Dask Cluster Creation."

**Reference issues/PRs**
Other issues that came up in this first usage are #25, #26 and #27.

### CODE REVIEW

**Comment by stsievert** [comment]
I did have one question on the "Using the start, status, and stop subcommands" subsection:  why can't I run `dask-chtc jupyter run lab` through tmux? That's how I normally persist Jupyter sessions on remote servers.

File: docs/source/jupyter.rst [hunk 1]
```diff
@@ -89,9 +86,8 @@ then the equivalent ``dask-chtc jupyter`` command would be
             http://localhost:8888/?token=fedee94f539b0beea492bb358d549ed79025b714f3b308c4
          or http://127.0.0.1:8888/?token=fedee94f539b0beea492bb358d549ed79025b714f3b308c4
 
-Note the HTCondor job events are mixed into the output stream.
-This is purely for diagnostic purposes;
-you may (for example) find them helpful if your notebook server job is
+HTCondor job events outputs some diagnostic information into this output
+stream. These messages may be helpful if your notebook server job is
```
Comment (JoshKarpel) [inline docs/source/jupyter.rst]:
```suggestion
Dask-CHTC mixes HTCondor job diagnostic information into the normal Jupyter output stream.
These messages may be helpful if your notebook server job is
```

File: docs/source/jupyter.rst [hunk 2]
```diff
@@ -55,24 +66,10 @@ The ``run`` subcommand is the simplest way to launch a Jupyter notebook server.
...
-            file:///home/karpel/.local/share/jupyter/runtime/nbserver-2176065-open.html
-        Or copy and paste one of these URLs:
-            http://localhost:8888/?token=9af0f23248bc26014727d1dc85b5fcc07f4803caf777f87a
-         or http://127.0.0.1:8888/?token=9af0f23248bc26014727d1dc85b5fcc07f4803caf777f87a
-
 
-then the equivalent ``dask-chtc jupyter`` command would be
+For example, if you normally start up a Jupyter Lab instance with ``jupyter
+lab``. The equivalent command for Dask-CHTC is ``dask-chtc jupyter run lab``
+command would be
```
Comment (JoshKarpel) [inline docs/source/jupyter.rst]: Grammar got a little messed up here. Maybe cut "command would be"?

File: docs/source/jupyter.rst [hunk 3]
```diff
@@ -22,6 +14,25 @@ Dask-CHTC provides a way to run a Jupyter notebook server on a CHTC submit node.
...
+
+
+.. _Jupyter install documentation: https://jupyter.org/install
+
+.. warning::
+
+   Jupyter must be installed. This amounts to following the
+   `Jupyter install documentation`_, which amounts to running
+   ``conda install -c conda-forge jupyterlab`` or adding `jupyter` to the
+   ``environment.yml`` file.
```
Comment (JoshKarpel) [inline docs/source/jupyter.rst]: Nitpick: can we not use "amounts" twice here? I'd also say that we should prefer `conda install` without `-c conda-forge` unless the package is actually not available in the standard channel.

File: dask_chtc/cli.py [hunk 4]
```diff
@@ -430,7 +430,8 @@ def contact_address(self) -> str:
     def start(self, jupyter_args: List[str]) -> "JupyterJobManager":
         if self.has_running_job():
             raise click.ClickException(
-                'You already have a running Jupyter notebook server; try the "status" subcommand to see it.'
+                "You already have a running Jupyter notebook server; "
+                'use "dask-chtc jupyter status" subcommand to see it\'s logs.'
```
Comment (JoshKarpel) [inline dask_chtc/cli.py:459]: I'm wondering whether we should go the full mile and just print the status as part of this message. It's not an expensive operation if we've already done the query (we should cache the query so we don't do it twice in rapid succession). If you don't want to touch that code, let me know and I'll do it in a followup PR (leaving this as-is).

**Review by JoshKarpel — CHANGES_REQUESTED** [review]
A few nitpicks, and one thought about the already-running-job message, but otherwise LGTM

**Comment by JoshKarpel** [comment]
> I did have one question on the "Using the start, status, and stop subcommands" subsection: why can't I run `dask-chtc jupyter run lab` through tmux? That's how I normally persist Jupyter sessions on remote servers.

Yep, that should work just fine. I want there to be a workflow that doesn't involve `tmux` for those who don't like it, but for those who do, `tmux` is a great way to make a detached session.

Do note that it won't be quite as protected as the fully-detached `dask-chtc jupyter start` job would be. CHTC admins do reserve the right to kill userland programs; if you accidentally start behaving badly in some part of your `tmux`, they might kill the whole thing as a safety measure.

File: docs/source/jupyter.rst (hunk 3 above)
Comment (stsievert) [reply docs/source/jupyter.rst]: https://jupyter.org/install recommend `conda install -c conda-forge jupyterlab`. However, it looks like it's available on the defaults channel (`conda install jupyter`). I'll make that edit.

File: dask_chtc/cli.py (hunk 4 above)
Comment (stsievert) [reply dask_chtc/cli.py:459]: > print the status as part of this message.

I think that's a good idea. I'd also add commands on how to stop/delete that Jupyter server.

**Comment by stsievert** [comment]
> CHTC admins do reserve the right to kill userland programs;

How can I be a good user so CHTC admins don't kill my jobs? I'm interested in running `jupyter lab` instead of `dask-chtc jupyter run lab` in a tmux session for maybe a couple days. What other downsides come from running `jupyter lab` instead of `dask-chtc jupyter run lab`?

**Comment by JoshKarpel** [comment]
> How can I be a good user so CHTC admins don't kill my jobs?

The main consideration is: try not to use large amounts of CPU, memory, or disk I/O on the submit node for longer than a few minutes at a time. The HTCondor daemons need those resources to serve other people's jobs. For a Dask workflow, this means making sure that you do all the heavy-duty computation through Dask, pulling only the minimal information back to the scheduler.

> What other downsides come from running `jupyter lab` instead of `dask-chtc jupyter run lab`?

Right now, very few (for you; there are some incidental benefits for us). But we're hoping to be able to enforce resource limits by doing this, which will make CHTC admins much happier with this workflow. As far as I know, there shouldn't be a difference in the behavior of Jupyter once you've got it running - let me know if there is, so that we can fix it :)

File: dask_chtc/cli.py (hunk 4 above)
Comment (JoshKarpel) [reply dask_chtc/cli.py:459]: @stsievert We'll probably have some new users trying things out in the next few days, so I'd like to get your improvements merged. Did you want to take a crack at this, or should I merge as-is?

Comment (stsievert) [reply dask_chtc/cli.py:459]: I haven't had time to try anything; feel free to merge if you're satisfied.

Comment (stsievert) [reply dask_chtc/cli.py:459]: And we can always add a follow-on PR.

Comment (JoshKarpel) [reply dask_chtc/cli.py:459]: :+1:

**Review by JoshKarpel — APPROVED** [review]

**Pull request merged**